File: main/src/ui/java/freedom/app/parser/SearchMessageParser.kt
package freedom.app.parser

object SearchMessageParser {

    // Wire format:
    //   Request:  "SRCH:REQ"
    //   Response: "SRCH:RESP:{json}"
    //   where json = {"searchable":true,"contacts":[{"name":"Alice","ref":"dGVzdA=="},...]}

    fun parse(raw: String, senderAddress: String?): ParsedMessage? {
        val body = raw.removePrefix("SRCH:")
        return when {
            body == "REQ" -> ParsedMessage(
                type          = MessageType.SEARCH_REQUEST,
                content       = "",
                senderAddress = senderAddress
            )
            body.startsWith("RESP:") -> ParsedMessage(
                type          = MessageType.SEARCH_RESPONSE,
                content       = body.removePrefix("RESP:"),
                senderAddress = senderAddress
            )
            else -> null
        }
    }
}

File: ios/Freedom/Core/Parser/SearchMessageParser.swift
// ═════════════════════════════════════════════════════════════════════════════
//  SearchMessageParser — Search / discovery messages
// ═════════════════════════════════════════════════════════════════════════════

import Foundation

enum SearchMessageParser {

    static func parse(_ raw: String, senderAddress: String? = nil) -> ParsedMessage? {
        let body = String(raw.dropFirst(5)) // drop "SRCH:"

        if body == "REQ" {
            return ParsedMessage(type: .SEARCH_REQUEST, content: "", senderAddress: senderAddress)
        }
        if body.hasPrefix("RESP:") {
            return ParsedMessage(type: .SEARCH_RESPONSE, content: String(body.dropFirst(5)), senderAddress: senderAddress)
        }

        return nil
    }
}
